### PR TITLE
Normalize workers assets path

### PR DESCRIPTION
**Pull request by WillTaylorDev**
Prevents same-schema redirect attacks by collapsing multiple slashes in the path.

- Tests
  - [ ] TODO (before merge)
  - [X] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [X] Not required because: n/a
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [X] Documentation not necessary because: n/a

### CODE REVIEW

File: packages/workers-shared/asset-worker/tests/handler.test.ts [hunk 1]
```diff
@@ -96,4 +96,46 @@ describe("[Asset Worker] `handleRequest`", () => {
...
+		expect(response.headers.get("location")).toBe("/abc/def/123/456");
+
+		response = await handleRequest(
+			new Request("https://example.com/%2fexample.com/"),
+			configuration,
+			exists,
+			getByETag
+		);
+
+		expect(response.status).toBe(307);
```
Comment (WillTaylorDev) [inline packages/workers-shared/asset-worker/tests/handler.test.ts:128]: These should be 200s. Will probably lose the location header when that updates.

File: fixtures/asset-config/redirects.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+		vi.mocked(applyConfigurationDefaults).mockImplementation(() => {
+			return {
+				html_handling: "none",
+				not_found_handling: "none",
+			};
+		});
+	});
+
+	it("returns 404 for non matched encoded url", async () => {
```
Comment (WalshyDev) [inline fixtures/asset-config/redirects.test.ts:48]: i'd probably have the normalisation stuff in it's own test to verify how it matches fully, bit weird it's in an encoded focused test

e.g. `/%2fexample.com/` -> 404
e.g. `/%2fblog/index.html` -> 200 `/blog/index.html`
etc

Comment (WillTaylorDev) [reply fixtures/asset-config/redirects.test.ts:48]: Added a url-normalization test file to test those specific cases. I think we still want the redirect tests, but you're right they can absolutely be split out.

**Review by CarmenPopoviciu — APPROVED** [review]

**Review by edmundhung — APPROVED** [review]

File: fixtures/asset-config/url-normalization.test.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		async (pathname: string) => {
+			if (fileList.has(pathname)) {
+				return pathname;
+			}
+		}
+	);
+};
+const BASE_URL = "http://example.com";
+
+describe("[Asset Worker] `test redirects`", () => {
```
Comment (petebacondarwin) [inline fixtures/asset-config/url-normalization.test.ts:23]: redirects? or rewrites?

Comment (WillTaylorDev) [reply fixtures/asset-config/url-normalization.test.ts:23]: Good catch, I'll update in my next PR shortly!

**Pull request merged**